kernel/sched.c

```diff
@@ -975,7 +975,15 @@ s32_t z_impl_k_sleep(int ms)
 {
 	s32_t ticks;
 
-	ticks = k_ms_to_ticks_ceil32(ms);
+	__ASSERT((ms >= 0) || (ms == K_FOREVER),
+		 "Only non-negative values are accepted.");
+	/* If asserts are disabled treat negative numbers as the past. */
+	if ((ms < 0) && (ms != K_FOREVER)) {
+		return 0;
+	}
+
+	ticks = (ms == K_FOREVER) ?
+			(INT32_MAX - _TICK_ALIGN) : k_ms_to_ticks_ceil32(ms);
```

Comment:
Changing my review -- @pabigot correctly pointed out that this will still expire within hours.
I feel we can treat this issue as a bug and not a feature, no need to rush it.